sdk/appconfiguration/azure-spring-cloud-feature-management/README.md

```diff
@@ -259,6 +259,171 @@ Options are available to customize how targeting evaluation is performed across
+          default: true
+          configuration-reference: "DiscountBanner:Big"
+          assignment-parameters:
+            ...
+        -
+          name: Small
+          default: false
+          configuration-reference: "DiscountBanner:Small"
+          assignment-parameters:
+            ...
+feature-variant:
+  DiscountBanner:
+    Big:
+      size: 400
+      color: "#BBB"
+    Small:
+      size: 150
+      color: "#DDD"
+```
+
+A Dynamic Feature has two main parts, the assigner and a list of variants. The assigner in this case `feature-management.DiscountBanner.assigner` which is pointing the `TargetingEvaluator` `Microsoft.Targeting` that will evaluate the list of variants and choose which configuration-reference will be used. Each variant has: a name, configuration-reference, and assignment-parameters. One of the listed variants needs to be configured as the default variant for the feature. The name field is meta data to reference the variant and just needs to be unique. The configuration-reference is a reference to the configurations defined under `feature-variant` in the configuration file. **Note:** the colon notation is required when denoting parts of the configuration. If it is needed multiple colons can be used to reference sub configurations. i.e. `DiscountBanner:Big:size` would return the `Integer` 400.
+
+The assignment-parameters takes in a `Map` which contains the evaluation criteria. This enables custom assigners to be created. In the case of `Microsoft.Targeting` it contains the `Audience` object.
+
+#### Targeting Evaluator
+
+`TargetingEvaluator` is a built-in assigner, which can be enabled by creating the `Microsoft.Targeting` Bean.
+
+```java
+@Bean(name = "Microsoft.Targeting")
+@RequestScope
+public TargetingEvaluator targetingEvaluator(TargetingContextImpl context) {
+    return new TargetingEvaluator(context, new TargetingEvaluationOptions().setIgnoreCase(true));
+}
+```
+
+`TargetingEvaluator` extends `TargetingFilter` so both can be used at the same time.
+
+The `Microsoft.Targeting` assigner uses three evaluation criteria to assign a variant; users, groups, and defaultRolloutPercentage. Assignment requires the definition of the `ITargetingContextAccessor` where a user defines how the user and group parameters are defined.
+
+If a users is listed in a variant then they are automatically assigned to that variant. If they are in more than one variant they are returned to the first listed variant.
+
+Groups have a rollout percentage. Using the user and group information, user id and group name. This is combined so users are bucketed into parts of the rollout. This enables a rollout where 10% of a group gets a variant, then when it increases to 50% that 10% still gets the variant with an additional 40% of the group now getting the variant. This can be done by changing:
+
+```yml
+Audience:
+  groups:
+    -
+        name: "Dev"
+        rolloutPercentage: 10
+  defaultRolloutPercentage: 0
+```
+
+to
+
+```yml
```

Comment:
Is it necessary? I feel that saying to change the percentage from 10 to 50 have the same affect.

Reply:
Do you mean the move one to the other example?

Reply:
I just meant that copying the configuration block and only tweaking the percentage might just be as easy as mentioning that the percentage needs to be updated.